src/v/storage/disk_log_impl.cc

```diff
@@ -129,6 +130,20 @@ ss::future<> disk_log_impl::close() {
             });
         });
     });
+
+    if (_segs.size()) {
+        auto clean_seg = _segs.back()->filename();
```

Comment:
I guess I'm wondering if this is going to always be clean. For example, right above this it seems like problems closing a segment are ignored, presumably because the assumption is that they are cleaned on startup?

```
        return ss::parallel_for_each(_segs, [](ss::lw_shared_ptr<segment>& h) {
            return h->close().handle_exception([h](std::exception_ptr e) {
                vlog(stlog.error, "Error closing segment:{} - {}", e, h);
```